### PR TITLE
Ncnews 11

### PR DESCRIPTION
**Pull request by majidfer**
Completed ncnews #11 with corresponding test

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -57,6 +57,94 @@ describe("GET /api/articles", () => {
...
+        body.articles.forEach((article) => {
+          expect(article).toEqual(
+            expect.objectContaining({
+              article_id: expect.any(Number),
+              title: expect.any(String),
+              topic: expect.any(String),
+              author: expect.any(String),
+              created_at: expect.any(String),
+              votes: expect.any(Number),
+              comment_count: expect.any(Number),
```
Comment (bainesface) [inline __tests__/app.test.js]: As you have previously used a forEach to assert the structure of the articles in another test, there is no need to do so for these query tests. Asserting that the articles have been sorted correctly will do nicely and will DRY up your test suite

File: __tests__/app.test.js [hunk 2]
```diff
@@ -57,6 +57,94 @@ describe("GET /api/articles", () => {
...
+              title: expect.any(String),
+              topic: "mitch",
+              author: expect.any(String),
+              created_at: expect.any(String),
+              votes: expect.any(Number),
+              comment_count: expect.any(Number),
+            })
+          );
+        });
+      });
```
Comment (bainesface) [inline __tests__/app.test.js:119]: Apart from this one. The forEach here is useful as we are checking every topic is relating to mitch. You could have just asserted this behaviour, however, and not the whole article object each time

File: models/articles.models.js [hunk 3]
```diff
@@ -1,22 +1,51 @@
...
+  ON articles.article_id = comments.article_id`;
+
+  if (topic) {
+    queryStr += ` WHERE topic = $1 
+                  GROUP BY articles.article_id`;
+    topicVal.push(topic);
+  } else {
+    queryStr += ` GROUP BY articles.article_id`;
+  }
+
```
Comment (bainesface) [inline models/articles.models.js:36]: To DRY up this code ... you could add the GROUP BY section of the query once, outside of the if else, as you are always wanting to add this section, regardless of whether a topic has been queried or not

File: models/articles.models.js [hunk 4]
```diff
@@ -1,22 +1,50 @@
...
+    "article_id",
+    "title",
+    "topic",
+    "author",
+    "body",
+    "created_at",
+    "votes",
+  ];
+  const validOrder = ["asc", "ASC", "desc", "DESC"];
+  // const validTopics =
```
Comment (mallionaire) [inline models/articles.models.js]: Please remove commented out code 😄

**Review by mallionaire — APPROVED** [review]
Lovely, just a little leftover commented out code. Please remove it then you're good to go

**Pull request merged**